hack/.clusterUtils

```diff
@@ -41,6 +41,45 @@ type: Opaque
 EOF
 }
 
+makeSecretForKubeconfig() {
+  local kubeconfig=$1
+  local clusterName=$2
+  local targetCluster=$3
+  local targetClusterName=${targetCluster:=$clusterName}
+
+  local tmpfile=$(mktemp /tmp/kubeconfig-internal.XXXXXX)
+  cp ${kubeconfig} ${tmpfile}
+
+  local server=$(kubectl --kubeconfig ${tmpfile} config view -o jsonpath="{$.clusters[?(@.name == '${clusterName}')].cluster.server}")
+  local caData=$(kubectl --kubeconfig ${tmpfile} config view --raw -o jsonpath="{$.clusters[?(@.name == '${clusterName}')].cluster.certificate-authority-data}")
+  local certData=$(kubectl --kubeconfig ${tmpfile} config view --raw -o jsonpath="{$.users[?(@.name == '${clusterName}')].user.client-certificate-data}")
+  local keyData=$(kubectl --kubeconfig ${tmpfile} config view --raw -o jsonpath="{$.users[?(@.name == '${clusterName}')].user.client-key-data}")
+  rm -f ${tmpfile}
+
+  cat <<EOF
+kind: Secret
+apiVersion: v1
+metadata:
+  name: ""
+  namespace: ""
+stringData:
+  config: >-
+    {
+      "tlsClientConfig":
+        {
+          "insecure": false,
+          "caData": "${caData}",
+          "certData": "${certData}",
+          "keyData": "${keyData}"
+        }
+    }
+  name: ${targetClusterName}
+  server: ${server}
+type: Opaque
+EOF
+
+}
+
```

Comment:
This looks almost identical to `makeSecretForCluster`, can one not call the other?

Reply:
Yes, I have it in mind to refactor it so `makeSecretForCluster` calls `makeSecretForKubeconfig`